core/src/components/datetime/datetime-util.ts

```diff
@@ -284,14 +285,23 @@ export const getLocalDateTime = (dateString: any = ''): Date => {
       date.getMilliseconds()
     )
   );
+
+  if (timeZone && timeZone.length > 0) {
```

Comment:
Doing the previous suggestion above, we can then do:
```suggestion
  if (timeZone !== undefined) {
```

No big performance benefit here, just a simpler way of doing it. Plus, if the user does input an empty string/invalid value for the timezone, the browser should throw an error. Thoughts?